### PR TITLE
MJPEG Rewrite

### PR DESCRIPTION
**Pull request by 132ikl**

### CODE REVIEW

File: opsi/modules/videoio/mjpeg.py [hunk 1]
```diff
@@ -178,216 +159,89 @@ async def send(self, data):
...
-        except (TypeError, ValueError):
-            LOGGER.error("Failed to parse URL parameters", exc_info=True)
-        return quality, fps, resolution
+            return int(self.request.query_params.get(name, default))
+        except ValueError:
+            LOGGER.error("Failed to parse URL parameter %s", name, exc_info=True)
+            return default
+
+    def get_params(self):
+        params = {"compression": 70, "fps": 30, "resolution": None}
```
Comment (dzil123) [inline opsi/modules/videoio/mjpeg.py]: Consider using a dataclass

File: opsi/modules/videoio/mjpeg.py [hunk 2]
```diff
@@ -178,216 +159,89 @@ async def send(self, data):
...
+        params["compression"] = self.int_param("compression", 30)
+        params["fps"] = self.int_param("fps", 30)
```
Comment (dzil123) [inline opsi/modules/videoio/mjpeg.py]: I don't like that you have defaults in two places, in the initial `params` declaration and in each `int_param`

File: opsi/modules/videoio/mjpeg.py [hunk 3]
```diff
@@ -178,216 +159,89 @@ async def send(self, data):
...
-            LOGGER.error("Failed to parse URL parameters", exc_info=True)
-        return quality, fps, resolution
+            return int(self.request.query_params.get(name, default))
+        except ValueError:
+            LOGGER.error("Failed to parse URL parameter %s", name, exc_info=True)
+            return default
+
+    def get_params(self):
+        params = {"compression": 70, "fps": 30, "resolution": None}
+        params["compression"] = self.int_param("compression", 30)
```
Comment (dzil123) [inline opsi/modules/videoio/mjpeg.py]: Also this default compression=30 is different from the other default compression=70

File: opsi/modules/videoio/mjpeg.py [hunk 4]
```diff
@@ -178,216 +159,89 @@ async def send(self, data):
...
-        self.queue = queue.Queue()
-        self.thread = AsyncThread(timeout=0.5, name="CameraSource")
-        self.snippets = []
-        self._img: Mat = None
-        self._shutdown = False
-
-    @property
-    def quality(self):
-        return self._quality
+        self.pendingFrame = False
```
Comment (dzil123) [inline opsi/modules/videoio/mjpeg.py:125]: `pendingFrame` is never used

**Review by dzil123 — CHANGES_REQUESTED** [review]

**Review by dzil123 — APPROVED** [review]

**Review by dzil123 — APPROVED** [review]

**Comment by dzil123** [comment]
Why did you do two reviews smh

**Comment by 132ikl** [comment]
i didn't :flushed:

**Comment by dzil123** [comment]
Oh wait that was me haha

**Pull request merged**